engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/FormDeploymentTest.java

```diff
@@ -0,0 +1,301 @@
+import org.junit.Test;
+
+public class FormDeploymentTest {
+
+  private static final String TEST_FORM_1 = "/form/test-form-1.form";
+  private static final String TEST_FORM_1_V2 = "/form/test-form-1_v2.form";
+  private static final String TEST_FORM_2 = "/form/test-form-2.form";
+  private static final String TEST_FORM_WITHOUT_ID = "/form/test-form_without_id.form";
+  private static final String TEST_FORM_WITH_BLANK_ID = "/form/test-form_with_blank_id.form";
+
+  @Rule public final EngineRule engine = EngineRule.singlePartition();
+
+  @Test
+  public void shouldDeployFormResource() {
+    // when
+    final var deploymentEvent = engine.deployment().withJsonClasspathResource(TEST_FORM_1).deploy();
+
+    // then
+    Assertions.assertThat(deploymentEvent)
+        .hasIntent(DeploymentIntent.CREATED)
+        .hasValueType(ValueType.DEPLOYMENT)
+        .hasRecordType(RecordType.EVENT);
+  }
+
+  @Test
+  public void shouldRejectWhenFormIdIsMissing() {
+    // when
+    final var deploymentEvent =
+        engine
+            .deployment()
+            .withJsonClasspathResource(TEST_FORM_WITHOUT_ID)
+            .expectRejection()
+            .deploy();
+
+    // then
+    Assertions.assertThat(deploymentEvent)
+        .hasIntent(DeploymentIntent.CREATE)
+        .hasRecordType(RecordType.COMMAND_REJECTION)
+        .hasRejectionType(RejectionType.INVALID_ARGUMENT);
+
+    assertThat(deploymentEvent.getRejectionReason())
+        .contains(String.format("Expected the form id to be present, but none given"));
+  }
+
+  @Test
+  public void shouldRejectWhenFormIdIsBlank() {
+    // when
+    final var deploymentEvent =
+        engine
+            .deployment()
+            .withJsonClasspathResource(TEST_FORM_WITH_BLANK_ID)
+            .expectRejection()
+            .deploy();
+
+    // then
+    Assertions.assertThat(deploymentEvent)
+        .hasIntent(DeploymentIntent.CREATE)
+        .hasRecordType(RecordType.COMMAND_REJECTION)
+        .hasRejectionType(RejectionType.INVALID_ARGUMENT);
+
+    assertThat(deploymentEvent.getRejectionReason())
+        .contains(String.format("Expected the form id to be filled, but it is blank"));
+  }
+
+  @Test
+  public void shouldWriteFormRecord() {
+    // when
+    engine.deployment().withJsonClasspathResource(TEST_FORM_1).deploy();
+
+    // then
+    final Record<Form> record = RecordingExporter.formRecords().getFirst();
+
+    Assertions.assertThat(record)
+        .hasIntent(FormIntent.CREATED)
+        .hasValueType(ValueType.FORM)
+        .hasRecordType(RecordType.EVENT);
+
+    assertThat(record.getKey()).isPositive();
+
+    final Form formRecord = record.getValue();
+    Assertions.assertThat(formRecord)
+        .hasFormId("Form_0w7r08e")
+        .hasResourceName(TEST_FORM_1)
+        .hasVersion(1);
+
+    assertThat(formRecord.getFormKey()).isPositive();
+    assertThat(formRecord.isDuplicate()).isFalse();
+  }
+
+  @Test
+  public void shouldDeployDuplicate() {
```

Comment:
🔧 Maybe the name of this test is a bit tricky, can you rename with something like `shouldDeployDuplicateInSeparateCommand`